.github/workflows/test.yml

```diff
@@ -16,8 +16,6 @@ jobs:
           fetch-depth: 1
       - name: Set up Python
         uses: actions/setup-python@v2
-        with:
-          python-version: '3.7.9'
```

Comment:
Wouldn't we need to specify Python `3.8.X` version here?

Reply:
@Sinclert if I understand it correctly one of the task in this PR takes care of setting up the python version so we no longer need this (@mnyrop please correct me if I am wrong)

Reply:
@ekate is correct. The GitHub action reads in the `.python-version` file if it exits

Reply:
I think it is not working... 😕 

It seems that, either the GH Action has a problem reading the `.python-version` file, or it takes the latest version upon a missing the `with: python-version: X.X.X` YAML specification.

Please, check how Python `3.9.7` (latest) was selected both in:
- The [PR branch CI workflow](https://github.com/nyudlts/ultraviolet/runs/3705380428?check_suite_focus=true#step:3:5).
- The [`master` branch CI workflow](https://github.com/nyudlts/ultraviolet/runs/3707271379?check_suite_focus=true#step:3:5).

Reply:
@mnyrop could you please address this

Reply:
@mnyrop I might be wrong but I think you still need `with` section but need to put a variable there 
```
 with:
          python-version: ${{ matrix.python-version }}
          architecture: x64
```

Reply:
The `${{ matrix.python-version }}` template is only going to get a value if the `matrix` YAML section is specified beforehand, as in [this example](https://github.com/psf/requests/blob/94f9f7357e50f6accf1ec1a87fd3223981e68003/.github/workflows/run-tests.yml#L11-L13). In our case, it is not defined.

Please, stick to the following for now:

```yaml
with:
    python-version: 3.8.12
```